### PR TITLE
feat: init pod controllers

### LINKED ISSUE #146: feat: synchronize pod to sub clusters
**What would you like to be added**:

Synchronize pod to sub clusters

**Why is this needed**:

### PR DESCRIPTION
**Pull request by OrangeBao**
#### What type of PR is this?

feat
#### What does this PR do?

 synchronize pod to sub clusters

#### Which issue(s) does this PR fix?

Fixes #146 
#### Screenshots (if applicable):

```release-note

```

#### Additional Comments:

```release-note
 synchronize pod to sub clusters

```

### CODE REVIEW

File: pkg/clustertree/cluster-manager/controllers/pod/pod_downstream_controller.go [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+	var pod corev1.Pod
+	if err := r.Get(ctx, request.NamespacedName, &pod); err != nil {
+		if errors.IsNotFound(err) {
+			leafPod := &corev1.Pod{}
+			err := r.LeafClient.Get(ctx, request.NamespacedName, leafPod)
+			if err != nil && errors.IsNotFound(err) {
+				return reconcile.Result{}, nil
+			}
+			// delete leaf pod
+			if err := r.DeletePodInLeaf(ctx, leafPod); err != nil {
```
Comment (xyz2277) [inline pkg/clustertree/cluster-manager/controllers/pod/pod_downstream_controller.go]: How about DeletePodInLeafCluster

Comment (OrangeBao) [reply pkg/clustertree/cluster-manager/controllers/pod/pod_downstream_controller.go]: ok, I get it.

**Pull request merged**